DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Centre/Reports/ReportsViewModel.cs

```diff
@@ -28,11 +30,44 @@ IEnumerable<EvaluationResponseBreakdown> evaluationResponseBreakdowns
 
     public class UsageStatsTableViewModel
     {
-        public UsageStatsTableViewModel(IEnumerable<PeriodOfActivity> activity)
+        public UsageStatsTableViewModel(IEnumerable<PeriodOfActivity> activity, DateTime startDate, DateTime endDate)
         {
-            Rows = activity.Reverse().Select(
-                p => new ActivityDataRowModel(p, DateHelper.GetFormatStringForDateInTable(p.DateInformation.Interval))
-            );
+            activity = activity.ToList();
+
+            if (activity.Count() <= 1)
+            {
+                Rows = activity.Select(
+                    p => new ActivityDataRowModel(p, DateHelper.StandardDateFormat, startDate, endDate)
+                );
+            }
+            else
+            {
+                var first = new[] { activity.First() }.Select(
+                    p => p.DateInformation.Interval == ReportInterval.Days
+                        ? new ActivityDataRowModel(
+                            p,
+                            DateHelper.GetFormatStringForDateInTable(p.DateInformation.Interval)
+                        )
+                        : new ActivityDataRowModel(p, DateHelper.StandardDateFormat, startDate, true)
+                );
+                var last = new[] { activity.Last() }.Select(
+                    p => p.DateInformation.Interval == ReportInterval.Days
+                        ? new ActivityDataRowModel(
+                            p,
+                            DateHelper.GetFormatStringForDateInTable(p.DateInformation.Interval)
+                        )
+                        : new ActivityDataRowModel(p, DateHelper.StandardDateFormat, endDate, false)
+                );

```

Comment:
Why do we need to use new lists of single items for these rather than just having them be converted directly?

Reply:
It was the first way that came to mind to deal with creating and concatenating the three components neatly, but now you've put a light on it I think just using prepend/append would be clearer.